src/main/resources/public/jsx/geo-dash-widget-editor.js

```diff
@@ -1,318 +1,344 @@
+             selectedDataType: "-1",
+             WidgetTitle: "",
+             imageCollection: "",
+             graphBand: "",
+             graphReducer: "Min",
+             imageParams: "",
+             dualLayer: false,
+             WidgetBaseMap: "osm",
+             startDate: "",
+             endDate: "",
+             startDate2: "",
+             endDate2: "",
+             widgetBands: "",
+             widgetMin: "",
+             widgetMax: "",
+             widgetCloudScore: "",
+             imageCollectionDual: "",
+             selectedDataTypeDual: "-1",
+             imageParamsDual: "",
+             startDateDual: "",
+             endDateDual: "",
+             widgetBandsDual: "",
+             widgetMinDual: "",
+             widgetMaxDual: "",
+             widgetCloudScoreDual: "",
+             FormReady: false,
+             wizardStep: 1,
+             pid: BasicLayout.getParameterByName("pid"),
+             institutionID: BasicLayout.getParameterByName("institutionId") != null? BasicLayout.getParameterByName("institutionId"): "1"
 
-
-    componentDidMount() {
-        const ref = this;
-        fetch(theURL + "/id/" + pid,)
-            .then(response => response.json())
-            .then(function(response){ref.setState({ dashboardID: response.dashboardID});  return response})
-            .then(function(response){
-                let theWidgets = response.widgets;
-                if(Array.isArray(theWidgets))
-                {
-                    return response;
-                }
-                else if(Array.isArray(eval(theWidgets))){
-                    response.widgets = eval(theWidgets);
-                }
-                else{
-                    response.widgets = [];
-                }
-                return response;
-            })
-            .then(data => data.widgets.map(function(widget){
-                if(widget.layout) {
-                    if (widget.layout.y == null) {
-                        widget.layout.y = 0;
-                    }
-                }
-                return widget;}))
+         };
+     }
+     componentDidMount() {
+         console.log("the institutionID is now set to: " + this.state.institutionID);
+         const ref = this;
+         fetch(this.props.theURI + "/id/" + this.state.pid,)
+             .then(response => response.json())
+             .then(function(response){ref.setState({ dashboardID: response.dashboardID});  return response;})
+             .then(function(response){
+                 let theWidgets = response.widgets;
+                 if(Array.isArray(theWidgets))
+                 {
+                     return response;
+                 }
+                 else if(Array.isArray(eval(theWidgets))){
+                     response.widgets = eval(theWidgets);
+                 }
+                 else{
+                     response.widgets = [];
+                 }
+                 return response;
```

Comment:
this seems unreachable

Reply:
The return after the else block still seem unreachable.  I think you can remove this line.